microraiden/setup.py

```diff
@@ -15,14 +15,43 @@
 import os
 from setuptools import Command
 from setuptools.command.build_py import build_py
-from microraiden.constants import MICRORAIDEN_VERSION
 
 DESCRIPTION = 'µRaiden is an off-chain, cheap, scalable and low-latency micropayment solution.'
+VERSION = open('microraiden/VERSION', 'r').read().strip()
+
+
+def read_version_from_git():
+    try:
+        import shlex
+        git_version, _ = subprocess.Popen(
+            shlex.split('git describe --tags'),
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE
+        ).communicate()
+        git_version = git_version.decode()
+        if git_version.startswith('v'):
+            git_version = git_version[1:]
+
+        git_version = git_version.strip()
+        # if this is has commits after the tag, it's a prerelease:
+        if git_version.count('-') == 2:
+            _, _, commit = git_version.split('-')
+            if commit.startswith('g'):
+                commit = commit[1:]
+            return '{}+git.r{}'.format(VERSION, commit)
```

Comment:
So, in case we have intermediary releases, the version would look like this: `0.2.0+git.rc80fc57`?
I personally like the `0.2.0-rc.2` pattern more - increasing the number makes the release order clearer.

Reply:
No, for pypi fix releases we'll bump patch version. Releases' version will never use `+git.*` suffix.
The `git` tag is only useful i.e. if someone uses the git code directly and files a bug report - then we know which commit he is on.

Release version bumping is simple - just branch out and do `bumpversion minor && git push` . This will automatically increment desired release number (major, minor or patch) and tag, update & commit all affected files (`setup.py` and `.bumpversion.cfg`)

Reply:
Sorry, when I said intermediary releases I was thinking of prereleases - between two patch versions (so, using`-rc.2` instead of `+git.rc80fc57`). I thought you also wanted to publish them. But yes, git tags are useful for debugging.